### DISCUSSION TITLE
Xdrip+ doesn't get any readings from Miaomiao1-Libre2

### DISCUSSION THREAD

**Discussion by fabriziusart** [NightscoutFoundation/xDrip#2749] [General]
Hi everyone, I need some help. My Xdrip+ app doesn't get any reading from MiaoMiao1 since I changed from Libre1 to Libre2. This is my second Libre2 sensor. I have already tried installing OOP2 app, both checked and unchecked options on the menu as I've read in a previous discussion, but still doesn't get any reading. Checking the log event I understand the app is connected, get the data but doesn't translate nor show them. Any suggestion please?
![Screenshot_20230326-160553](https://user-images.githubusercontent.com/43249546/227781699-c3b6438c-b192-4f88-88c5-591c1ecdb537.png)
![Screenshot_20230326-160648](https://user-images.githubusercontent.com/43249546/227781712-954c10ce-3d3a-43ed-964d-2671fa499cd1.png)

**Comment by Navid200** [comment]
@psonnera

**Comment by psonnera** [comment]
@fabriziusart first please confirm this is an Europe model of the Libre 2.
OOP2 doesn't answer. Can you confirm you downloaded the correct version https://xdrip.readthedocs.io/en/latest/use/OOP/#oop2 and you configured it correctly? Also check it's not being optimized (killed) by energy savings https://xdrip.readthedocs.io/en/latest/troubleshoot/savings/.

**Reply by fabriziusart** [reply]
Hi, thanks for your reply.
Yes it's an Europe model and I have the last version of OOP2, also the battery settings are right. To make sure I uninstalled and reinstalled both XDRIP+ and OOP2 apps and rechecked the settings.

**Comment by tzachi-dar** [comment]
You need to update the fw version of the miaomiao to the latest version.
https://miaomiaoreader.medium.com/how-to-make-libre-us-14-days-sensors-work-with-miaomiao2-on-xdrip-35b431a40940

**Reply by fabriziusart** [reply]
Hi, thanks for your reply.
I tried uploading the update on the Miaomiao and failed with "error 4096 DFU DEVICE DISCONNECTED".
I stumbled upon this tutorial several days ago and failed at that time too. I have already tried resetting and recharging the miaomiao.

**Reply by psonnera** [reply]
Did you "forget device" in xDrip+ system status before? https://xdrip.readthedocs.io/en/latest/troubleshoot/bridgeFW/#miaomiao

**Reply by fabriziusart** [reply]
I did forget device and tried uploading again but it doesn't work. The upload fails every time. Also installing the tomato app of the vendor doesn't get better. It connects to my miaomiao, tries several seconds to upload and fails. I now believe it is my device which has some problem. I also tried on another phone all of these steps with the same result. Thanks to all for the support till now. Anything else I could try?

**Reply by psonnera** [reply]
That doesn't sound good. Retry with Tomato and keep the device on a sensor.

**Reply by fabriziusart** [reply]
Hi, after dozens of failed attempts I've found a solution in this tutorial [https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip]. I installed xDrip-plus-20230328-940daa7.apk. I hope it's not a temporary solution because my Miaomiao still doesn't update the new firmware. Anyway it's working perfectly. Many thanks for the precious help and quick reply. @psonnera  @tzachi-dar @Navid200
